ethereum/blockwatch/client.go

```diff
@@ -22,29 +26,64 @@ type Client interface {
 
 // RpcClient is a Client for fetching Ethereum blocks from a specific JSON-RPC endpoint.
 type RpcClient struct {
+	rpcClient      *rpc.Client
 	client         *ethclient.Client
 	requestTimeout time.Duration
 }
 
 // NewRpcClient returns a new Client for fetching Ethereum blocks using the given
 // ethclient.Client.
-func NewRpcClient(ethClient *ethclient.Client, requestTimeout time.Duration) (*RpcClient, error) {
-	return &RpcClient{ethClient, requestTimeout}, nil
+func NewRpcClient(rpcURL string, requestTimeout time.Duration) (*RpcClient, error) {
+	ethClient, err := ethclient.Dial(rpcURL)
+	if err != nil {
+		return nil, err
+	}
+	rpcClient, err := rpc.Dial(rpcURL)
+	if err != nil {
+		return nil, err
+	}
+	return &RpcClient{rpcClient, ethClient, requestTimeout}, nil
+}
+
+type GetBlockByNumberResponse struct {
+	Hash       common.Hash `json:"hash"`
```

Comment:
For future reference: if you're just unmarshaling, you don't need to add these `json` struct tags. The `encoding/json` package will account for the fact that cases might be different in Go vs. JSON and will automatically unmarshal `hash` into `Hash`. It also doesn't break anything to include them so it's fine for now.